### PR TITLE
lab 05 updated, first commit

### LINKED ISSUE #449: Lab 05 Stringer
/* Issue to be closed when pull request approved and merged */

Create an executable go program in directory 05_stringer/marchitect
Make the IPAddr type implement fmt.Stringer to print the address as a dotted quad

### PR DESCRIPTION
**Pull request by Marchitect**
Fixes #449 

Review of colleague's PR #

#### Changes proposed in this PR:

- added test case for main()

### CODE REVIEW

**Review by joel00wood — COMMENTED** [review]
Only thing I can really suggest is more test cases. Currently your tests confirm that the output of main() is as expected but it has no breadth of how the `String()` override behaves on your `IPAddr` data type. 

At least worth confirming behaviour for:
- Empty `IPAddr{}`
- Partially Implemented `IPAddr{1,2}`
- Fully Implemented `IPAddr{1,2,3,4}`

**Comment by Marchitect** [comment]
> Only thing I can really suggest is more test cases. Currently your tests confirm that the output of main() is as expected but it has no breadth of how the `String()` override behaves on your `IPAddr` data type.
> 
> At least worth confirming behaviour for:
> 
> * Empty `IPAddr{}`
> * Partially Implemented `IPAddr{1,2}`
> * Fully Implemented `IPAddr{1,2,3,4}`

Thank you for your feedback :) kindly help check if the mentioned suggestion is now resolved?

**Comment by joel00wood** [comment]
Yep looks good to me!

File: 05_stringer/marchitect/main.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+var out io.Writer = os.Stdout
+
+type IPAddr [4]byte
+
+func (ip IPAddr) String() string {
+	return fmt.Sprintf("%d.%d.%d.%d", ip[0], ip[1], ip[2], ip[3])
+}
+
+// Make the IPAddr type implement fmt.Stringer to print the address as a dotted quad
```
Comment (juliaogris) [inline 05_stringer/marchitect/main.go:17]: Comment without context - is it for `main`?

File: 05_stringer/marchitect/main.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (juliaogris) [inline 05_stringer/marchitect/main.go:18]: Unnecessary blank line.
No need for blank lines at beginning or end of block.
Use blank lines between type, method and function definitions and within a block for logical separation.

**Review by juliaogris — CHANGES_REQUESTED** [review]
Add link to your review of colleague's PR please.  
Please [fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages) with `git rebase -i HASH`.

**Pull request closed without merging**